packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,213 @@
+
+### Anatomy 
+
+Shadow DOM:
+```html
+<template id=WaferMapTemplate>
+	<div class="WaferMapContainer">
+		<svg>
+			<g class="zoomContainer">
+				<g>
+				<circle></circle>
+				<circle></circle>
+				<rect></rect>
+				</g>
+			</g>
+		</svg>
+		<div class=WaferMapArea>
+			<canvas/>
+			<svg>
+				<rect>
+				</rect>
+			</svg>
+		</div>
+	</div>
+</template>
+```
+The first SVG element with the "zoomContainer" class attribute contains two circles and a rectangle.
+* The first circle represents the outline of the wafer map component.
+* The second circle, together with the rectangle is part of a solution which creates the illusion that there is a notch present on the wafer map frame.
+
+The second SVG element with the "WaferMapArea" class attribute represents the area where the dies get rendered.
+The rectangle present here is used as the "highlight" which allows us to identify individual dies in the canvas.
+
+- Slot Names: (none)
+- Host Classes: (none)
+- Slotted Content/Slotted Classes: (none)
+- CSS Parts: (none)
+
+### Angular integration
+
+The `nimble-wafer-map` component does not take user input directly from the webpage where it's embedded. All the inputs must be provided through the constructor of the component so in the current case there is no point using ControlValueAccessors.
+
+### Blazor integration
+
+Blazor support is out of scope for the current implementation.
+
+### Visual Appearance
+
+For the initial component, we are going with the original wafer map appearance. 
+![WaferMap Example](./Resources/wafer_map_example.png)
+
+## Implementation
+
+As mentioned above, this is not a component that we start from scratch, but a visualization that already exists which is outdated and not futureproof. 
+As a first step, we are working on the "bloat removal" which is basically the removal of unwanted but exiting features (legend, tooltip etc.) and dependencies.
+
+Original component: [wafermap](https://dev.azure.com/ni/DevCentral/_git/op-web-portal?path=/libraries/wafermap&version=GBfeature/microServicesSplit)\
+Once the "bloat removal" is finished, the next step is to adapt the current 
+
+### Code submission strategy
+
+In order to make the process of submitting the re-written wafer map component easier for us and for the code reviewers as well, we propose the following bottom-up approach:
+* As an initial step, we submit a skeleton of the `nimble-wafer-map` component into the main branch. This will serve as a starting point on which the complete functionality will be built on.
+* Once the skeleton is in place, we start to work on and submit individual pieces of the functionality so it's not very overwhelming to review the code.
+
+### States
+
+We are not planning to allow the user to generate an “empty" wafer map, so we must ensure that data is provided when instantiating a new `nimble-wafer-map` component.
```

Comment:
As suggested in other comments, I think ultimately we _will_ have a concept of an "empty" wafer-map, as we won't be able to insist that data is present before the component is in the DOM. Thus, I think the question is do we need to come up with a design for an empty wafer-map?

Reply:
Good point. In that specific case we could draw just the outline of the wafer map (the circle with the notch) without the dies in it. The concept of "not having an empty wafer map" I was referring to the fact that we would like not allow the user to create an empty visualization, which will be filled afterwards with data. (for example having an "Add data" button which allows to upload a CSV representing the wafer map data.). That is why I mentioned that it's mandatory to provide the data when instantiating a new wafer map component. But in case the data is not correct and no dies can be drawn into the wafer map area, I agree, that an empty wafer map should be displayed.